### ISSUE TITLE
Eviction manager's observed memory.available is vastly different from /proc/meminfo MemAvailable

### ISSUE THREAD

**Issue by mtaufen** [kubernetes/kubernetes#28619]
I'm working on a node e2e memory eviction test, and even though I set the aggressive hard eviction threshold of ` --eviction-hard memory.available<2.5Gi`, I noticed that my pods were chewing up memory without getting evicted, so I started digging.

I logged the `observations` made from the `synchronize` function in `kubernetes/pkg/kubelet/eviction/eviction_manager.go`, with `glog.Infof("eviction manager: observations: %s", strings.Replace(spew.Sdump(observations), "\n", " ", -1))`, and I saw the following:

`eviction manager: observations: (eviction.signalObservations) (len=1) {  (eviction.Signal) (len=16) "memory.available": (*resource.Quantity)(0xc820d61940)(3377236Ki) }`

sampling `/proc/meminfo` about two seconds later reveals:
```
MemTotal:        3801016 kB
MemFree:          826748 kB
MemAvailable:    1213620 kB
```

Assuming kB and Ki are equivalent units (they seem to both be kibibytes), that's a 2163616Ki difference in reporting between `memory.available` and `MemAvailable`.

My test collectively chews up memory in steps of about 40MB every 10 seconds, so they could show up as different, but shouldn't be THAT different.

Tagging @vishh @derekwaynecarr

**Comment by vishh** [comment]
cc @derekwaynecarr

**Comment by vishh** [comment]
@mtaufen and I looked at this issue together briefly today and we suspect that cAdvisor is not accounting for tmpfs as part of `working_set`. We need to isolate and confirm this issue. 
@derekwaynecarr since your day starts earlier, do take a look at cAdvisor tmpfs handling if you get a chance.

**Comment by vishh** [comment]
Bytes written to tmpfs are being accounted as part of the overall usage of the cgroups. cAdvisor attempts to ignore inactive files, but weirdly, all writes to tmpfs from inside a memcg is getting accounted as inactive file usage instead of active file usage.
Ideally, we want `working-set` to match the kernel logic [here](http://lxr.free-electrons.com/source/mm/oom_kill.c#L187).
I will continue looking at this issue tomorrow.

cc @timstclair

**Comment by vishh** [comment]
It looks like tmpfs is being charged correctly in newer kernels, 4.5+. @adityakali helped confirm on GCI and CoreOS. 
At the least, we know that our working_set logic is sane. 
Next step is to find out if this bug exists in any of the other supported distros.

**Comment by vishh** [comment]
`working_set` computed in cAdvisor is incorrect. It needs to take into account the status of swap. If swap is disabled, it should not assume that `inactive_anon` pages are reclaimable. 
The behavior when swap is enabled is TBD. Given that we recommend users to disable swap, `working_set` computation with swap enabled is not a high priority issue.

**Comment by vishh** [comment]
@mtaufen is working on a cAdvisor fix which needs to be cherry-picked into v1.3 branch.
cc @timstclair ^^
